media/capture/content/android/java/src/org/chromium/media/ScreenOffloading.java

```diff
@@ -0,0 +1,327 @@
+            ByteBuffer buf, int rowStride, int left, int top, int width, int height,
+            long timestamp);
+
+    private native void nativeOnI420FrameAvailable(long nativeScreenCaptureMachineAndroid,
+            ByteBuffer yBuffer, int yStride, ByteBuffer uBuffer, ByteBuffer vBuffer,
+            int uvRowStride, int uvPixelStride, int left, int top, int width, int height,
+            long timestamp);
+
+    // Method for ScreenOffloading implementations to notify activity result.
+    private native void nativeOnActivityResult(
+            long nativeScreenCaptureMachineAndroid, boolean result);
+
+    // Method for ScreenOffloading implementations to notify orientation change.
+    private native void nativeOnOrientationChange(
+            long nativeScreenCaptureMachineAndroid, int rotation);
+
+    private void handleImage(ImageWrapper imageWrapper) {
+        if (mIsStopped == true) return;
+
+        switch (imageWrapper.format) {
+            case PixelFormat.RGBA_8888:
+                nativeOnRGBAFrameAvailable(mNativeScreenCaptureMachineAndroid,
+                        ScreenCaptureService.sVideoBuffer[0], imageWrapper.rowStride[0],
+                        imageWrapper.rectLeft, imageWrapper.rectTop, imageWrapper.rectWidth,
+                        imageWrapper.rectHeight, imageWrapper.timestamp);
+                break;
+            case ImageFormat.YUV_420_888:
+                // The pixel stride of Y plane is always 1. The U/V planes are guaranteed
+                // to have the same row stride and pixel stride.
+                nativeOnI420FrameAvailable(mNativeScreenCaptureMachineAndroid,
+                        ScreenCaptureService.sVideoBuffer[0], imageWrapper.rowStride[0],
+                        ScreenCaptureService.sVideoBuffer[1], ScreenCaptureService.sVideoBuffer[2],
+                        imageWrapper.rowStride[1], imageWrapper.pixelStride, imageWrapper.rectLeft,
+                        imageWrapper.rectTop, imageWrapper.rectWidth, imageWrapper.rectHeight,
+                        imageWrapper.timestamp);
+                break;
+            default:
+                Log.e(TAG, "Unexpected image format: " + imageWrapper.format);
+                throw new IllegalStateException();
+        }
+    }
+
+    private void handleAudio(int bytesRead) {
+        if (mIsStopped == true) return;
+        if (AudioRecordBridge.getAudioRecord() != null) {
+            byte[] temp = new byte[ScreenCaptureService.sAudioBuffer.remaining()];
+            synchronized (ScreenCaptureService.sAudioSync) {
+                ScreenCaptureService.sAudioBuffer.position(0);
+                ScreenCaptureService.sAudioBuffer.get(temp);
+
+                ScreenCaptureService.sAudioSync.notifyAll();
+            }
+
+            AudioRecordBridge.getAudioRecord().sAudioBuffer.clear();
+            AudioRecordBridge.getAudioRecord().sAudioBuffer.put(temp);
```

Comment:
In |put| function, |temp|'s data is copied to |sAudioBuffer|'s buffer?
If |sAudioBuffer| has its own buffer, |temp| needs to be allocated every time?